docs: Enhance Javadoc documentation across core components

vero-code opened the pull request:
### Description

This pull request significantly enhances and standardizes the Javadoc documentation for key classes and methods across 21 files within the `core` package, specifically focusing on the LLM flow and Event handling components.

### Key Changes & Improvements

* **Comprehensive Javadoc Coverage:** Added or expanded Javadoc comments for critical methods and classes, ensuring clear explanations of their purpose, parameters, return values, and potential exceptions. This includes, but is not limited to, methods in `LlmFlow` (e.g., `callLlm`, `handleBeforeModelCallback`, `handleAfterModelCallback`, `runOneStep`, `run`, `runLive`), `AgentTransfer`, `ExampleUtils`, and various callback-related classes.
* **Standardization & Readability:**
    * Converted numerous single-line comments into proper Javadoc blocks where appropriate, improving consistency and automated documentation generation.
    * Addressed instances where comments were missing or represented by empty quotes, filling in necessary documentation.
    * Ensured consistent use of Javadoc tags (e.g., `@param`, `@return`, `@throws`, `{@link}`) and standard formatting for better readability and navigation within IDEs and generated documentation.

### Rationale & Value Add

This work aims to significantly improve the overall readability, maintainability, and comprehensibility of the codebase. Enhanced Javadoc documentation:
* **Facilitates Onboarding:** Makes it easier for new developers to understand complex flows and APIs quickly.
* **Reduces Errors:** Provides clearer guidance on method usage, reducing potential for misuse.
* **Boosts Maintainability:** Simplifies future debugging, refactoring, and feature development by providing immediate context for code elements.

This contribution aligns with best practices for code documentation and directly supports the long-term health and usability of the `adk` project.

vorburger reviewed (approved):
LGTM.

shukladivyansh commented:
Hi Veronika!

Thanks for the PR. Really appreciate the efforts to improve the docs.

We won't be able to merge this PR for a few reasons. 

1. Too noisy - We want our docstrings to be concise and meaningful
2. Alignment with [adk-python](https://github.com/google/adk-python) - We lean on adk-python for being the source of truth and the docstrings shouldn't deviate from them
3. AI Generated - This PR feels AI generated. It's ok to generate the first draft using AI but we won't accept docs without further revision and human input.

vero-code commented:
Thank you so much for the detailed feedback! I've simplified the Javadoc, removed redundant parts, and aligned it with the adk-python style where applicable. I'm happy to address any further comments.

vorburger reviewed (approved):
LGTM.

Poggecci commented on core/src/main/java/com/google/adk/agents/BaseAgent.java:
```diff
@@ -58,6 +58,15 @@ public abstract class BaseAgent {
+   * @param name Unique agent name.
+   * @param description Agent purpose.
+   * @param subAgents Agents managed by this agent.
+   * @param beforeAgentCallback Callbacks before agent execution.
```
Could you update this to mention how callbacks are called in order until one doesn't return None? Same for afterAgentCallback
https://github.com/google/adk-python/blob/9597a446fdec63ad9e4c2692d6966b14f80ff8e2/src/google/adk/agents/base_agent.py#L98

Poggecci commented on core/src/main/java/com/google/adk/agents/BaseAgent.java:
```diff
@@ -164,6 +173,12 @@ public Optional<List<AfterAgentCallback>> afterAgentCallback() {
   }
 
+  /**
+   * Creates a context based on the parent.
```
Please add that this is a shallow copy of the parent context with the agent properly being set to this instance

Poggecci reviewed (changes requested):
Thank you very much for this contribution! Requested a few changes but after that, LGTM!
